addons/website_sale/static/src/js/website_sale.js

```diff
@@ -209,6 +236,17 @@ publicWidget.registry.WebsiteSale = publicWidget.Widget.extend(VariantMixin, {
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * Set the url hash from the checked inputs and
+     * the selected options in the parent provided
+     * @private
+     */
+    _setUrlHash: function ($parent) {
+        var combinations = $parent.find('input.js_variant_change:checked, select.js_variant_change option:selected').map(function (idx, elem) {
+            return $(elem).data('value_id');
+        }).toArray();
+        history.replaceState(undefined, undefined, '#' + combinations);
```

Comment:
Can't you just write on the hash directly instead of a strange call with two leading `undefined` values ? Also I would `join` the array with `,` explicitly instead of relying on the array `toString` method. Also use `_.map` instead of `$.map`.

Reply:
If i replace the hash like this: ```window.location.hash = '#' + combinations.join(',');```
The history is impacted and if we want to browse the previous page we will have to go trough all the previous variants combinations.
history.replaceState() replaces the current state in the history wich is I think the expected behavior
Also with undefined as first and second parameter I make sure not to override the history.state and the document.title. Alternatively I can do it this way : ```history.replaceState(history.state, document.title, '#' + combinations.join(','));```

Reply:
Mmh cool, did not know that :+1: 
Still use `_.map` and an explicity `join` though :wink:

Reply:
I have made those changes